gui/src/2D/controls/selector.ts

```diff
@@ -0,0 +1,383 @@
+        this._panel.top = 5;
+        this._panel.left = 5;
+        
+        this._addGroupHeader(groups[0].name);
+        this._addGroup(groups[0]);
+        for(var i = 1; i < groups.length; i++) {
+            this._addSpacer(groups[i].name);
+            this._addGroup(groups[i]);
+        }				
+        this.addControl(this._panel);
+        
+        
+        return this;
+    }
+    
+    /** Adds the given group to the SelectionPanel
+     * @param group, the SelectionGroup to be added
+    */
+    protected _addGroup(group: SelectorGroup) {
+        if(group.type == "R") {
+            for(var i = 0; i < group.selectors.length; i++) {
+                this._addRadio(group.selectors[i].text, group.selectors[i].nb, group.name, group.selectors[i].func, group.selectors[i].checked);
+            }
+        }
+        else if(group.type == "S") {
+            this._panel.width = 1;
+            for(var i = 0; i < group.selectors.length; i++) {
+                this._addSldr(group.selectors[i].text, group.selectors[i].func, group.selectors[i].unit, group.selectors[i].min, group.selectors[i].max, group.selectors[i].value, group.selectors[i].onVal);
+            }
+        }
+        else {
+            for(var i = 0; i < group.selectors.length; i++) {
+                this._addCheckbox(group.selectors[i].text, group.selectors[i].func, group.selectors[i].checked);
+            }
+        }
+    }
+    
+    /** Adds a heading to the group
+     * @param name is used as heading
+     */
+    protected _addGroupHeader(name: string) {
+        var groupHeading = new TextBlock("groupHead", name);
+        groupHeading.width = 0.9;
+        groupHeading.height = "30px";
+        groupHeading.textWrapping = true;
+        groupHeading.color = "black";
+        groupHeading.horizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+        groupHeading.textHorizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+        groupHeading.left = "2px";
+        this._panel.addControl(groupHeading);
+    }
+    
+    /** Adds a bar between groups
+     * @param name is used as heading for the group after the bar
+     */
+    protected _addSpacer(name: string) {
+        var separator = new Rectangle();
+        separator.width = 0.9;
+        separator.height = "2px";
+        separator.horizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+        separator.background = "#364249";
+        separator.color = "#364249";
+        this._panel.addControl(separator);
+    
+        this._addGroupHeader(name);
+    }
+    
+    /** Adds a checkbox as a control
+     * @param text is the label for the selector
+     * @param func is the function called when the Selector is checked
+     * @param checked is true when Selector is checked
+     */
+    protected _addCheckbox(text: string, func: (s: any)=>any, checked: boolean) {
+        var checked = checked || false;
+        var button = new Checkbox();
+        button.width = "20px";

```

Comment:
same comment: we need to expose headers and controls directly

Reply:
Same reply